### PR TITLE
Assign portfolio change

### PR DESCRIPTION
**Pull request by patrick-weiss**
Changed the assign_portfolio() functions and associated references in the text.

### CODE REVIEW

File: univariate-portfolio-sorts.qmd [hunk 1]
```diff
@@ -102,27 +102,34 @@ The results indicate that we cannot reject the null hypothesis of average return
...
   breakpoints <- data |>
-    reframe(
-      breakpoint = quantile(
-        {{ var }},
-        probs = seq(0, 1, length.out = n_portfolios + 1),
-        na.rm = TRUE)
-      ) |>
-    pull(breakpoint) |>
-    as.numeric()
+    pull(all_of(sorting_variable)) |>
```
Comment (christophscheuch) [inline univariate-portfolio-sorts.qmd:115]: Würde lieber curly-curly behalten. Geht nicht auch pull({{sorting_variable}})?

Comment (patrick-weiss) [reply univariate-portfolio-sorts.qmd:115]: curly-curly does work (of course, the input has to be changed as well). I found all_of() simpler to understand though, but I guess some learning might be good. @voigtstefan What do you think?

Comment (christophscheuch) [reply univariate-portfolio-sorts.qmd:115]: Ja, in der selection ist all_of() schon praktisch, aber wir haben curly-curly eingeführt, um zu zeigen, was man damit machen kann. all_of() ersetzt sicher nicht curly-curly in allen use cases. Daher würde ich curly-curly behalten.

Comment (patrick-weiss) [reply univariate-portfolio-sorts.qmd]: Reintroduced the curly-curly operator in Commit https://github.com/tidy-finance/website/commit/a24d695f1a6e44a194c3fc0a6fc58882eec789b9. Should I also add this to the changelog?

File: univariate-portfolio-sorts.qmd [hunk 2]
```diff
@@ -102,27 +102,34 @@ The results indicate that we cannot reject the null hypothesis of average return
...
+Now we take portfolio sorts to the next level. We want to be able to sort stocks into an arbitrary number of portfolios. For this case, functional programming is very handy: we employ `all_of` inside a `pull()` to give us flexibility concerning which variable to use for the sorting, denoted by `sorting_variable`. We use `quantile()` to compute breakpoints for `n_portfolios`. Then, we assign portfolios to stocks using the `findInterval()` function. The output of the following function is a new column that contains the number of the portfolio to which a stock belongs.\index{Functional programming} 
+
+In some applications, the variable used for the sorting might be clustered (e.g., at a lower bound of 0). Then, multiple breakpoints may be identical, leading to empty portfolios. This can be easily identified in the output by checking if the number of the portfolios' constituents is constant across portfolios. Similarly, some portfolios have a very small number of stocks, e.g., at the beginning of the sample.
```
[the remark this answers is not in the document]
Comment (patrick-weiss) [reply univariate-portfolio-sorts.qmd]: If you have 25% of a sorting variable equal to zero, then the first and second breakpoints are 0, but with findInterval() your second portfolio will be empty. This is not appealing.

Suggested a clarification in Commit https://github.com/tidy-finance/website/commit/35229d71cad7b99c9b0cbb522be9f652370c7c3f.

[the remark this answers is not in the document]
Comment (christophscheuch) [reply univariate-portfolio-sorts.qmd]: But what is the correct sorting if the first 25% of stocks have the same sorting value and you want to sort them into deciles? Randomly put stocks into the first 3 deciles? I guess a lower number of portfolios should be the solution or putting all the stocks in the same portfolio and you just have less portfolios (as findInterval does).

File: univariate-portfolio-sorts.qmd (hunk 1 above)
Comment (christophscheuch) [reply univariate-portfolio-sorts.qmd]: What do you mean by "this"? The change in assign_portfolio, yes. The reitroduction of curly-curly, no.

Comment (patrick-weiss) [reply univariate-portfolio-sorts.qmd]: this = the commit https://github.com/tidy-finance/website/commit/a24d695f1a6e44a194c3fc0a6fc58882eec789b9. Or should I just mention the original commit?

File: univariate-portfolio-sorts.qmd (hunk 2 above)
[the remark this answers is not in the document]
Comment (patrick-weiss) [reply univariate-portfolio-sorts.qmd]: "Correct" is a word that can hardly describe such an issue. Most commonly authors put all 0's in one portfolio and split the remaining cross section into (n_portfolios - 1). I have not seen a random assignment.

File: univariate-portfolio-sorts.qmd (hunk 1 above)
Comment (christophscheuch) [reply univariate-portfolio-sorts.qmd]: I'd just link this pull request and briefly summarize what has happened.

File: univariate-portfolio-sorts.qmd [hunk 3]
```diff
@@ -102,27 +102,34 @@ The results indicate that we cannot reject the null hypothesis of average return
...
+Now we take portfolio sorts to the next level. We want to be able to sort stocks into an arbitrary number of portfolios. For this case, functional programming is very handy: we employ the [curly-curly](https://www.tidyverse.org/blog/2019/06/rlang-0-4-0/#a-simpler-interpolation-pattern-with-)-operator to give us flexibility concerning which variable to use for the sorting, denoted by `sorting_variable`.\index{Curly-curly} We use `quantile()` to compute breakpoints for `n_portfolios`. Then, we assign portfolios to stocks using the `findInterval()` function. The output of the following function is a new column that contains the number of the portfolio to which a stock belongs.\index{Functional programming} 
+
+In some applications, the variable used for the sorting might be clustered (e.g., at a lower bound of 0). Then, multiple breakpoints may be identical, leading to empty portfolios. Similarly, some portfolios might have a very small number of stocks at the beginning of the sample. Such portfolio imbalances can be easily identified in the output by checking the number of each portfolio's constituents over time. Additionally, consider investigating the breakpoints over time.
```
Comment (voigtstefan) [inline univariate-portfolio-sorts.qmd:107]: Please remove this "easily identified" or clarify what could be analyzed here. Based on my understanding of the topic, I suggest something along the lines of: 
"Cases, where the number of portfolio constituents differs substantially due to the distribution of the characteristics, require careful consideration and, depending on the application, do prevent meaningful sorting mechanisms."

Comment (christophscheuch) [reply univariate-portfolio-sorts.qmd:107]: Bin bei Stefan, dass ich hier warnender auftreten würde, aber leicht anders:
"Cases, where the number of portfolio constituents differs substantially due to the distribution of the characteristics, require careful consideration and, depending on the application, might require customized sorting approaches."

Comment (patrick-weiss) [reply univariate-portfolio-sorts.qmd]: Small alteration in Commit https://github.com/tidy-finance/website/pull/38/commits/9bcec97dee89f9c727ae15011d5da36c67db8e14

File: univariate-portfolio-sorts.qmd (hunk 1 above)
Comment (patrick-weiss) [reply univariate-portfolio-sorts.qmd]: - [May 13, 2023, Commit 1232646:](https://github.com/tidy-finance/website/commit/1232646023dfd8b551d9d047d40c310550564b6b) We changed the assign_portfolio()-functions in Chapters [Univariate Portfolio Sorts](https://www.tidy-finance.org/univariate-portfolio-sorts.html), [Size Sorts and p-Hacking](https://www.tidy-finance.org/size-sorts-and-p-hacking.html), [Value and Bivariate Sorts](https://www.tidy-finance.org/value-and-bivariate-sorts.html), and [Replicating Fama and French Factors](https://www.tidy-finance.org/replicating-fama-and-french-factors.html). Additionally, we added a small explanation to potential issues with the function for clustered sorting variables in Chaptes [Univariate Portfolio Sorts](https://www.tidy-finance.org/univariate-portfolio-sorts.html) (see [Commit 24f9e60](https://github.com/tidy-finance/website/commit/24f9e6048f4f23a37e508fd3b98a8d73854e0dad)).

**Review by voigtstefan — APPROVED** [review]
Approved

**Pull request merged**